munit-hedgehog/shared/src/main/scala/munit/HedgehogSuite.scala

```diff
@@ -0,0 +1,115 @@
+package munit
+
+import hedgehog._
+import hedgehog.core.{PropertyConfig, PropertyT, Seed}
+import hedgehog.predef.Monad
+
+import munit.internal.Compat._
+
+/**
+ * Provides the ability to write property based tests using the Hedgehog library.
+ *
+ * Properties are defined by calling one of the various `property` or `propertyF`
+ * methods and passing a `PropertyT[Result]` or `PropertyT[F[Result]]` respectively.
+ * For example:
+ * {{{
+ * property("additive identity") {
+ *   Gen.int(Range.linearFrom(0, Int.MinValue, Int.MaxValue)).forAll.map { n =>
+ *     assertEquals(n + 0, n)
+ *   }
+ * }
+ * }}}
+ *
+ * Note both Hedgehog `Result` and munit assertions are supported.
+ */
+trait HedgehogSuite extends FunSuite {
+
+  /**
+   * Provides the default configuration for running property tests.
+   * This can be overriden to change the default or can be changed on
+   * a test-by-test basis by using an overload of `property` and `propertyF`.
+   */
+  lazy val propertyConfig: PropertyConfig = PropertyConfig.default
+  lazy val propertySeed: Long = System.currentTimeMillis()
+
+  def property(
+      name: String
+  )(prop: PropertyT[Result])(implicit loc: Location): Unit =
+    property(name, propertyConfig)(prop)
+
+  def property(
+      options: TestOptions
+  )(prop: PropertyT[Result])(implicit loc: Location): Unit =
+    property(options, propertyConfig)(prop)
+
+  def property(
+      name: String,
+      config: PropertyConfig
```

Comment:
To reduce the number of method overloads, would it make sense to pass the property config via tags instead?
```scala
property("name".tag(customPropertyConfig))
```

Reply:
I looked in to that but note doing so would require the ability to convert property config to/from String, unless I'm misunderstanding.

Reply:
BTW, I used `Conversion` to avoid the need for importing `language.implicitConversions` when working with Dotty. Maybe we should do the same for `Conversion[String, TestOptions]`?

Reply:
It's possible to create a tag that contains custom values like this
```scala
case class HedgehogConfig(config: PropertyConfig) extends Tag("HedgehogConfig")
```
See https://scalameta.org/munit/docs/tests.html#customize-evaluation-of-tests-with-tags

I didn't know about the `Conversion` trick to avoid the import. That's handy. However, I think it's good to keep the `name: String` method overload to make the sources more approachable for Scala beginners

Reply:
Ah, cool, I'll update the overloads.

Re: `Conversion`, I'm getting warnings with munit 0.7.1 when calling `test("name".ignore)`. I was wondering if we could use `Conversion` to avoid such warnings. I'll open a separate issue on that.

Reply:
Please do!